backend/Dockerfile

```diff
@@ -39,6 +39,7 @@ RUN apt-get -y update && apt-get install -y \
     binutils-aarch64-linux-gnu \
     curl \
     gcc-mips-linux-gnu \
+    qemu-user\
```

Comment:
is there a way to bundle this into the compiler_dir / download.py such that it's an optional dependency? It'll bump the docker image by ~200mb.. which I guess won't be an issue at all if/when compilers are all split out.. but yeh, just asking the question.

```sh
root@a17720767740:/# apt-get install qemu-user
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  binfmt-support libglib2.0-0 libglib2.0-data libicu66 libpipeline1 libxml2 qemu-user-binfmt shared-mime-info tzdata xdg-user-dirs
Suggested packages:
  sudo
The following NEW packages will be installed:
  binfmt-support libglib2.0-0 libglib2.0-data libicu66 libpipeline1 libxml2 qemu-user qemu-user-binfmt shared-mime-info tzdata xdg-user-dirs
0 upgraded, 11 newly installed, 0 to remove and 6 not upgraded.
Need to get 27.4 MB of archives.
After this operation, 219 MB of additional disk space will be used.
```

Reply:
in fact... this doesnt even install qemu-irix?
```
root@a17720767740:/# ll /usr/bin/qemu-*  
-rwxr-xr-x 1 root root 5706200 Dec  8 09:15 /usr/bin/qemu-aarch64*
-rwxr-xr-x 1 root root 5710296 Dec  8 09:15 /usr/bin/qemu-aarch64_be*
-rwxr-xr-x 1 root root 4147912 Dec  8 09:15 /usr/bin/qemu-alpha*
-rwxr-xr-x 1 root root 4956856 Dec  8 09:15 /usr/bin/qemu-arm*
-rwxr-xr-x 1 root root 4965048 Dec  8 09:15 /usr/bin/qemu-armeb*
-rwxr-xr-x 1 root root 4136296 Dec  8 09:15 /usr/bin/qemu-cris*
-rwxr-xr-x 1 root root 4205480 Dec  8 09:15 /usr/bin/qemu-hppa*
-rwxr-xr-x 1 root root 4631976 Dec  8 09:15 /usr/bin/qemu-i386*
-rwxr-xr-x 1 root root 4463432 Dec  8 09:15 /usr/bin/qemu-m68k*
-rwxr-xr-x 1 root root 4096072 Dec  8 09:15 /usr/bin/qemu-microblaze*
-rwxr-xr-x 1 root root 4087880 Dec  8 09:15 /usr/bin/qemu-microblazeel*
-rwxr-xr-x 1 root root 5777432 Dec  8 09:15 /usr/bin/qemu-mips*
-rwxr-xr-x 1 root root 5830456 Dec  8 09:15 /usr/bin/qemu-mips64*
-rwxr-xr-x 1 root root 5822264 Dec  8 09:15 /usr/bin/qemu-mips64el*
-rwxr-xr-x 1 root root 5773336 Dec  8 09:15 /usr/bin/qemu-mipsel*
-rwxr-xr-x 1 root root 5834552 Dec  8 09:15 /usr/bin/qemu-mipsn32*
-rwxr-xr-x 1 root root 5826360 Dec  8 09:15 /usr/bin/qemu-mipsn32el*
-rwxr-xr-x 1 root root 4105608 Dec  8 09:15 /usr/bin/qemu-nios2*
-rwxr-xr-x 1 root root 4078184 Dec  8 09:15 /usr/bin/qemu-or1k*
-rwxr-xr-x 1 root root 5388968 Dec  8 09:15 /usr/bin/qemu-ppc*
-rwxr-xr-x 1 root root 5452360 Dec  8 09:15 /usr/bin/qemu-ppc64*
-rwxr-xr-x 1 root root 5476936 Dec  8 09:15 /usr/bin/qemu-ppc64abi32*
-rwxr-xr-x 1 root root 5448264 Dec  8 09:15 /usr/bin/qemu-ppc64le*
-rwxr-xr-x 1 root root 4268264 Dec  8 09:15 /usr/bin/qemu-riscv32*
-rwxr-xr-x 1 root root 4268264 Dec  8 09:15 /usr/bin/qemu-riscv64*
-rwxr-xr-x 1 root root 4659208 Dec  8 09:15 /usr/bin/qemu-s390x*
-rwxr-xr-x 1 root root 4139976 Dec  8 09:15 /usr/bin/qemu-sh4*
-rwxr-xr-x 1 root root 4144072 Dec  8 09:15 /usr/bin/qemu-sh4eb*
-rwxr-xr-x 1 root root 4379400 Dec  8 09:15 /usr/bin/qemu-sparc*
-rwxr-xr-x 1 root root 4453896 Dec  8 09:15 /usr/bin/qemu-sparc32plus*
-rwxr-xr-x 1 root root 4453896 Dec  8 09:15 /usr/bin/qemu-sparc64*
-rwxr-xr-x 1 root root 4041320 Dec  8 09:15 /usr/bin/qemu-tilegx*
-rwxr-xr-x 1 root root 4631752 Dec  8 09:15 /usr/bin/qemu-x86_64*
-rwxr-xr-x 1 root root 6002312 Dec  8 09:15 /usr/bin/qemu-xtensa*
-rwxr-xr-x 1 root root 5971624 Dec  8 09:15 /usr/bin/qemu-xtensaeb*
```

Reply:
needing to install this giant dependency kinda makes this "just use qemu-irix" idea a lot less appealing imo

Reply:
bump. this again is why we wanted to use qemu-irix - otherwiwe we're introducing a huge dependency. I thought qemu-user wasn't needed?

Reply:
It's not necessary, I just forgot to remove that line